core-processor/src/processing.rs

```diff
@@ -224,14 +226,15 @@ impl ProcessErrorCase {
         }
     }
 
-    pub fn to_reason_and_payload(&self) -> (ErrorReplyReason, Vec<u8>) {
+    pub fn to_reason_and_payload(&self) -> (ErrorReplyReason, Option<Payload>) {
         match self {
             ProcessErrorCase::ExecutionFailed(
                 reason @ ActorExecutionErrorReplyReason::Trap(TrapExplanation::Panic(buf)),
-            ) => (reason.as_simple().into(), buf.to_vec()),
+            ) => (reason.as_simple().into(), Some(buf.inner().clone())),
             this => {
                 let (reason, payload) = this.to_reason_and_payload_str();
-                (reason, payload.into_bytes())
+                let payload = payload.into_bytes().try_into().ok();
```

Comment:
isn't it always ok?

Reply:
It might be but further code have `unreachable` + `log` branch